src/promptflow-tools/promptflow/tools/exception.py

```diff
@@ -58,6 +58,18 @@ def to_openai_error_message(e: Exception) -> str:
               "please make sure you have proper role assignment on your azure openai resource. You can refer to " \
               "https://learn.microsoft.com/en-us/azure/ai-services/openai/how-to/role-based-access-control"
         return f"OpenAI API hits {ex_type}: {msg}"
+    elif "messages with role 'tool' must be a response" in error_message:
+        msg = "Invalid parameter: messages with role 'tool' must be a response to a preceding message with " \
+              "'tool_calls'. Please make sure your chat prompt includes 'tool_calls' under the role 'assistant'. " \
+              "You could refer to guideline at https://aka.ms/pfdoc/chat-prompt"
+        return f"OpenAI API hits {ex_type}: {msg}"
+    elif (
```

Comment:
Looks we can merge the two cases into one condition, as they share the same tsg?

Reply:
updated